sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -259,6 +276,10 @@ private case class MySQLDialect() extends JdbcDialect with SQLConfHelper with No
     case StringType => Option(JdbcType("LONGTEXT", java.sql.Types.LONGVARCHAR))
     case ByteType => Option(JdbcType("TINYINT", java.sql.Types.TINYINT))
     case ShortType => Option(JdbcType("SMALLINT", java.sql.Types.SMALLINT))
+    // Because MySQL 5.7 only supports DOUBLE in the CAST function, and DOUBLE is a
```

Comment:
Is MySQL 5.7 the only exception？

Reply:
Yes, here are snapshots about how different versions behave.
**MySQL 5.7**
![image](https://github.com/user-attachments/assets/51c0fd05-0969-4550-953c-1740e5e7b29e)
**MySQL 8.0**
![image](https://github.com/user-attachments/assets/1a7e5f64-4f95-4764-9eec-f02d72072b77)
**MySQL 9.1**
<img width="478" alt="image" src="https://github.com/user-attachments/assets/cd97ab59-5241-44f2-9de8-2e5a960e21cb" />

Reply:
Even though 5.7 is officially end of life, there are still many people using it. Therefore I added support to MySQL 5.7. But I totally understand if we want to get rid of legacy and it's ok to remove this rewrite.

Reply:
OK, thank you for the infomation